pkg/component/registrycaches/registry_caches.go

```diff
@@ -335,13 +334,16 @@ func (r *registryCaches) registryCacheObjects(ctx context.Context, cache *regist
 			Name:      name,
 			Namespace: metav1.NamespaceSystem,
 			Labels:    registryutils.GetLabels(name, upstreamLabel),
+			// StatefulSets need to be recreated due to the removal of the `spec.serviceName` field and the addition of the `spec.revisionHistoryLimit` field.
+			// TODO(dimitar-kostadinov): Remove the `DeleteOnInvalidUpdate` annotation in the v0.19.0 release.
```

Comment:
Nit: In Gardener, we usually keep such migration codes for at least 3 releases.

From https://github.com/gardener/gardener/blob/v1.128.0/docs/development/process.md#deprecations-and-backwards-compatibility:
> In case you have to run migration code (_which is mostly internal_), please add a [TODO statement](#todo-statements) to remove it only after **3 releases**.

```suggestion
			// TODO(dimitar-kostadinov): Remove the `DeleteOnInvalidUpdate` annotation in the v0.21.0 release.
```